firestore/CMakeLists.txt

```diff
@@ -179,6 +179,7 @@ target_link_libraries(firebase_firestore
   PUBLIC
     firebase_app
     firebase_auth
+    absl_variant
```

Comment:
Pretty sure this should only be needed with firestore_core, so it'd be better to do it in the thing below. There are some platforms that firestore_core is not used, which that FIRESTORE_USE_EXTERNAL_CMAKE_BUILD flag captures

Reply:
Makes sense.
Had to create another PR because this branch had some stuff accidentally merged in.
https://github.com/firebase/firebase-cpp-sdk/pull/133
Closing this one.